lib/puppet/defaults.rb

```diff
@@ -1758,8 +1758,16 @@ def self.initialize_default_settings!(settings)
         for the node stored in puppetdb are current. However, this will double the fact
         submission load on puppetdb, so it is disabled by default.",
     },
+    :publicdir => {
+      :default  => nil,
```

Comment:
This should default to `"$vardir/public"`, so then you won't need to modify `install.rb` or other places that deal with "root" directories.